lib/emque/consuming/adapters/rabbit_mq/worker.rb

```diff
@@ -73,11 +77,54 @@ def process_message(delivery_info, metadata, payload)
               )
               ::Emque::Consuming::Consumer.new.consume(:process, message)
               channel.ack(delivery_info.delivery_tag)
-            rescue StandardError
-              channel.nack(delivery_info.delivery_tag)
+            rescue StandardError => ex
+              if enable_delayed_message
+                publish_to_delayed_message(delivery_info, metadata, payload)
+              else
+                channel.nack(delivery_info.delivery_tag)
+              end
             end
           end
 
+          def publish_to_delayed_message(delivery_info, metadata, payload)
+            headers = metadata[:headers] || {}
+            headers["x-delay"] = 1000
+            channel.ack(delivery_info.delivery_tag)
```

Comment:
the ack does seem safe, the headers assignment shouldn't fail, therefore the ack will be the first think called and feels like a pretty good guarantee that it'll happen, even if the publish fails. What about if the ack succeeds and the publish fails, that's the worst case, the message is lost at that point? Maybe we should ack after the publish, ala a transaction?